monai/transforms/post/array.py

```diff
@@ -46,3 +49,93 @@ def __call__(self, img, to_onehot=None, num_classes=None):
+
+    For example:
+    Use KeepLargestConnectedComponent with applied_values=[1], connectivity=1
+
+       [1, 0, 0]         [0, 0, 0]
+       [0, 1, 1]    =>   [0, 1 ,1]
+       [0, 1, 1]         [0, 1, 1]
+
+    Use KeepLargestConnectedComponent with applied_values[1, 2], independent=False, connectivity=1
+
+      [0, 0, 1, 0 ,0]           [0, 0, 1, 0 ,0]
+      [0, 2, 1, 1 ,1]           [0, 2, 1, 1 ,1]
+      [1, 2, 1, 0 ,0]    =>     [1, 2, 1, 0 ,0]
+      [1, 2, 0, 1 ,0]           [1, 2, 0, 0 ,0]
+      [2, 2, 0, 0 ,2]           [2, 2, 0, 0 ,0]
+
+    Use KeepLargestConnectedComponent with applied_values[1, 2], independent=True, connectivity=1
+
+      [0, 0, 1, 0 ,0]           [0, 0, 1, 0 ,0]
+      [0, 2, 1, 1 ,1]           [0, 2, 1, 1 ,1]
+      [1, 2, 1, 0 ,0]    =>     [0, 2, 1, 0 ,0]
+      [1, 2, 0, 1 ,0]           [0, 2, 0, 0 ,0]
+      [2, 2, 0, 0 ,2]           [2, 2, 0, 0 ,0]
+
+    Use KeepLargestConnectedComponent with applied_values[1, 2], independent=False, connectivity=2
+
+      [0, 0, 1, 0 ,0]           [0, 0, 1, 0 ,0]
+      [0, 2, 1, 1 ,1]           [0, 2, 1, 1 ,1]
+      [1, 2, 1, 0 ,0]    =>     [1, 2, 1, 0 ,0]
+      [1, 2, 0, 1 ,0]           [1, 2, 0, 1 ,0]
+      [2, 2, 0, 0 ,2]           [2, 2, 0, 0 ,2]
+
+    """
+
+    def __init__(self, applied_values, independent=True, background=0, connectivity=None):
+        """
+        Args:
+            applied_values (list or tuple of int): number list for applying the connected component on.
+                The pixel whose value is not in this list will remain unchanged.
+            independent (bool): consider several labels as a whole or independent, default is `True`.
+                Example use case would be segment label 1 is liver and label 2 is liver tumor, in that case
+                you want this "independent" to be specified as False.
+            background (int): Background pixel value. The over-segmented pixels will be set as this value.
+            connectivity (int): Maximum number of orthogonal hops to consider a pixel/voxel as a neighbor.
+                Accepted values are ranging from  1 to input.ndim. If ``None``, a full
+                connectivity of ``input.ndim`` is used.
+        """
+        super().__init__()
+        self.applied_values = applied_values
+        self.independent = independent
+        self.background = background
+        self.connectivity = connectivity
+        if background in applied_values:
+            raise ValueError("Background pixel can't be in applied_values")
+
+    def __call__(self, img):
+        """
+        Args:
+            img: shape must be (num_channels, spatial_dim1[, spatial_dim2, ...]).
+
+        Returns:
+            A PyTorch Tensor with shape (1, spatial_dim1[, spatial_dim2, ...]).
+        """
+        if img.size()[0] == 1:
+            img = torch.squeeze(img, dim=0)
+        else:
+            img = torch.argmax(img, dim=0)
```

Comment:
Let's don't do `Argmax`, it's the work of another post transform `ConvertForMetrics`, will do later.
And we will try to support One-Hot format data(multi-labels task) after this PR merged, which doesn't expect Argmax operation for channels.

Reply:
If we don’t do argmax here, we don’t support multichannel input in this transform then. 
Because if input is a multichannel tensor that represents the output, I think each channel will corresponds to one label. And usually they are probability

If we don’t want to do extra thing in this transform, I think we can add in docstring saying we just support one channel where the pixel value in it is the prediction label.

Or you think there might be cases where each channel is having their own prediction label of multiple classes?

Reply:
1. If the model output is for "multi-labels" task and you hard-code to do `Argmax` will cause wrong output.
2. And even if the model output only has 1 channel but not rounds to 0 or 1, this transform also can't handle it. Need to add sigmoid first and round values by (> 0.5).

So in order to handle all kinds of model outputs, I am developing a separate post-transform, and can make a **post-transform chain** with this KeepLargestConnectedComponnent and SplitChannel transforms. The signature of that transform looks like:
```py
class ConvertForMetrics(Transform):
    def __init__(
        self,
        y_pred,
        y,
        add_sigmoid=False,
        add_argmax=False,
        round_values=False,
        to_onehot_y_pred=False,
        to_onehot_y=False,
        logit_thresh=0.5
    )
```
 Let's keep this PR simple to support only 1 channel first, then let's work together to submit another PR to support multi-channels case for this issue: https://github.com/Project-MONAI/MONAI/issues/420
What do you think?

Thanks.

Reply:
I will go with the first, which is remove argmax and add docstring "Expected input data should have only 1 channel and the values correspond to expected labels."